docs/DeveloperGuide.adoc

```diff
@@ -844,14 +859,46 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Delete student
+
+*MSS*
+
+1.  User requests to add a lesson
+2.  Scheduler shows a list of free slots
+3.  User requests to add a lesson to a slot
+4.  Scheduler adds that lesson to that slot
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. The timetable is empty.
++
+Use case ends.
+[none]
+* 2b. The timetable is full.
++
+** 3b1. Scheduler shows an error message.
+
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. Scheduler shows an error message.
++
+Use case resumes at step 2.
+
 _{More to be added}_
 
 [appendix]
 == Non Functional Requirements
 
 .  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `1.8.0_60` or higher installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+.  Should be able to hold up to 1000 students without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  Should receive feedback after executing commands.
```

Comment:
This is a functional requirement. A similar NFR would be "should give feedback to the user within X seconds of executing a command".